### PR TITLE
add-kb-pegasus

### PR DESCRIPTION
**Pull request by melonbred**
Add Pegasus, a 40% keyboard project.

## Description

Add Pegasus, a 40% keyboard project.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pegasus/pegasus.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    return process_record_user(keycode, record);
+}
+
+bool led_update_kb(led_t led_state) {
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+    return led_update_user(led_state);
+}
+*/
```
Comment (zvecr) [inline keyboards/pegasus/pegasus.c:50]:
```suggestion
```

File: keyboards/pegasus/config.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
+#define DEVICE_VER 0x0001
+#define MANUFACTURER melonbred
+#define PRODUCT Pegasus
+#define DESCRIPTION 12.75u WKL 40% Keyboard
```
Comment (zvecr) [inline keyboards/pegasus/config.h:28]:
```suggestion
```

File: keyboards/pegasus/config.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
+
+
```
Comment (zvecr) [inline keyboards/pegasus/config.h:87]:
```suggestion
```

File: keyboards/pegasus/keymaps/vial/config.h [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
Comment (zvecr) [inline keyboards/pegasus/keymaps/vial/config.h:1]: `vial` keymap has to be completely removed

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, this wont actually compile due to missing `BOOTLOADER` in `rules.mk`

File: keyboards/pegasus/keymaps/vial/config.h (hunk 4 above)
Comment (melonbred) [reply keyboards/pegasus/keymaps/vial/config.h]: Thanks for the quick review! I removed `vial` keymap, included `BOOTLOADER` to rules.mk, and committed your suggestions. Please let me know if there might be anything else I should revise. Thanks again!

File: keyboards/pegasus/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    }
+    return false;
+}
+
+
+
```
Comment (drashna) [inline keyboards/pegasus/keymaps/default/keymap.c:71]:
```suggestion
```

File: keyboards/pegasus/keymaps/split/keymap.c [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    }
+    return false;
+}
+
+
+
```
Comment (drashna) [inline keyboards/pegasus/keymaps/split/keymap.c:72]:
```suggestion
```

File: keyboards/pegasus/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+![pegasus](https://i.imgur.com/6MKJN7w.jpg)
+
+Pegasus is a 40% keyboard with a 12.75u "WKL" layout with 0.75u blockers. 
+
+* Keyboard Maintainer: [melonbred](https://github.com/melonbred)
+* Hardware Supported: The PCBs, controllers supported
+* Hardware Availability: Links to where you can find this hardware
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/pegasus/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/pegasus/rules.mk [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+WS2812_DRIVER = no
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/pegasus/rules.mk:22]:
```suggestion
BOOTMAGIC_ENABLE = yes  # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no    # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no     # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes       # Enable N-Key Rollover
BACKLIGHT_ENABLE = no
RGBLIGHT_ENABLE = no
ENCODER_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/pegasus/pegasus.c [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "pegasus.h"
```
Comment (drashna) [inline keyboards/pegasus/pegasus.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "pegasus.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
	    if (clockwise) {
	        tap_code(KC_VOLU);
	    } else {
	        tap_code(KC_VOLD);
	    }
	} else if (index == 1) {
	    if (clockwise) {
	        tap_code(KC_PGDN);
	    } else {
	        tap_code(KC_PGUP);
	    }
    }	
    return true;
}
#endif
```

File: keyboards/pegasus/readme.md (hunk 7 above)
Comment (melonbred) [reply keyboards/pegasus/readme.md:11]: Thank you for your review! I have added this to readme.md

File: keyboards/pegasus/readme.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# Pegasus
+
+![pegasus](https://i.imgur.com/6MKJN7w.jpg)
```
Comment (fauxpark) [inline keyboards/pegasus/readme.md:3]:
```suggestion
![pegasus](https://i.imgur.com/6MKJN7wh.jpg)
```

File: keyboards/pegasus/rules.mk [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+BOOTMAGIC_ENABLE = yes  # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no    # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no     # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes       # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/pegasus/rules.mk:20]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/pegasus/rules.mk (hunk 11 above)
Comment (melonbred) [reply keyboards/pegasus/rules.mk:20]: `rules.mk` was requested to be revised by @drashna yesterday. I'm happy to commit this suggestion, but for my own knowledge, is this suggestion the preferred formatting/wording moving forward?

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**